syntax = "proto3";

message ProtoTypeCard {
    string code = 1;
    sint32 id = 2;
}

message EventInitialHand {
    sint32 count = 1;
    repeated ProtoTypeCard cards = 2;
    bool first_turn = 3;
}

message EventMoveCard {
    string code = 1;
    sint32 card_id = 2;
    string start_zone = 3;
    sint32 start_pos = 4;
    string target_zone = 5;
    sint32 target_pos = 6;
    repeated ProtoTypeCard markers = 7;
    sint32 marker_pos = 8;
    bool insert_facedown = 9;
    string purpose = 10;
}

message EventPlayCard {
    sint32 card_id = 1;
    sint32 hand_pos = 2;
    sint32 stage_pos = 3;
    string code = 4;
}

message EventSwitchStagePositions {
    sint32 stage_pos_from = 1;
    sint32 stage_pos_to = 2;
}

message EventClockToWr {}

message EventStartMoveLog {}
message EventEndMoveLog {
    bool is_confirmed = 1;
}
